cmake/linker/ld/target.cmake

```diff
@@ -121,7 +128,7 @@ function(toolchain_ld_link_elf)
     ${ARGN}                                                   # input args to parse
   )
 
-  if(${CMAKE_LINKER} STREQUAL "${CROSS_COMPILE}ld.bfd")
+  if(${CMAKE_LINKER} MATCHES "${CROSS_COMPILE}ld.bfd$")
```

Comment:
note: `CROSS_COMPILE` may be an absolute path, so if this change is done because you want to handle cases where `CROSS_COMPILE` is just a string and `CMAKE_LINKER` is an absolute path, then new surprises could be lurking in the dark.

Could you comment a bit on your observations / reasons for the change ?
Also consider adding a comment in the CMake code, just to on safe side in case other starts to modify this.

Reply:
`${CMAKE_LINKER}` always comes back as full path to the executable on my tests. So a straight `STREQUAL` would never match. Let me try using `get_filename_component()` to extract the file name for comparison.